test/e2e/storage/flexvolume.go

```diff
@@ -117,11 +124,26 @@ func uninstallFlex(node *v1.Node, vendor, driver string) {
 	sshAndLog(cmd, host)
 }
 
-func getFlexDir(master bool, vendor, driver string) string {
+func getFlexDir(c clientset.Interface, node *v1.Node, vendor, driver string) string {
 	volumePluginDir := defaultVolumePluginDir
 	if framework.ProviderIs("gce") {
-		if (master && framework.MasterOSDistroIs("gci")) || (!master && framework.NodeOSDistroIs("gci")) {
-			volumePluginDir = gciVolumePluginDir
+		if node == nil && framework.MasterOSDistroIs("gci") {
+			v, err := getMasterVersion(c)
+			if err != nil {
+				framework.Failf("Error getting master version: %v", err)
+			}
+
+			if v.AtLeast(versionutil.MustParseGeneric(gciVolumePluginDirVersion)) {
+				volumePluginDir = gciVolumePluginDir
+			} else {
+				volumePluginDir = gciVolumePluginDirLegacy
+			}
+		} else if node != nil && framework.NodeOSDistroIs("gci") {
```

Comment:
Do we have tests where the master is on a higher version than the nodes?

Reply:
Not sure, I did a quick search and didn't find any tests that get master version. The cluster upgrade test has a function to verify the master version, and my implementation is based on that

Reply:
Ok, I was worried about a case where it would install the flex driver on the node based off of the master version.  But it looks like that's not the case.